pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/json/ImmutableJsonIndexReader.java

```diff
@@ -101,6 +103,23 @@ public MutableRoaringBitmap getMatchingDocIds(String filterString) {
     }
   }
 
+  public ImmutableRoaringBitmap getDocIds(int dictId) {
+    ImmutableRoaringBitmap flattenedDocIds = _invertedIndex.getDocIds(dictId);
+    MutableRoaringBitmap result = new MutableRoaringBitmap();
+    flattenedDocIds.stream().forEach(f -> result.add(getDocId(f)));
+    return result;
+  }
+
+  public Pair<Object[], Integer> getValues(String key) {
+    int startIndex = _dictionary.indexOf(key + JsonIndexCreator.KEY_VALUE_SEPARATOR);
```

Comment:
I realized that the current json indexing store the whole key/value pair as a sorted string table without compression even though the key is the common prefix when I see this two lines of code :)